src/EnergyPlus/InputProcessing/InputProcessor.cc

```diff
@@ -701,26 +701,22 @@ int InputProcessor::getIntFieldValue(json const &ep_object, json const &schema_o
     auto it = ep_object.find(fieldName);
     if (it != ep_object.end()) {
         auto const &field_value = it.value();
-        if (field_value.is_number()) {
-            if (field_value.is_number_integer()) {
-                value = field_value.get<std::int64_t>();
-            } else {
-                value = nint(field_value.get<double>());
-            }
-        } else {
-            bool is_empty = field_value.get<std::string>().empty();
-            if (is_empty) {
-                isDefaulted = findDefault(defaultValue, schema_field_obj);
-            } else {
-                value = DataGlobalConstants::AutoCalculate; // autosize and autocalculate
+        if (field_value.is_number_integer()) {
+            value = field_value.get<std::int64_t>();
+        } else if (field_value.is_number()) {
+            // This is a developer error, floating point numbers should not be retrieved this way. If this field
+            // really is an int then the input processor will have forced it to be an integer.
+            assert(!field_value.is_number());
```

Comment:
Reworked this function so that it doesn't round, that's no longer legal using the direct epJSON input. It's a developer error to call this function on a field that is stored as a floating point, probably need a better way to enforce that than `assert`.

Reply:
Hopefully our test coverage hits this enough so that we get the assert during development.  :shrug: